### PR TITLE
[BFCL] Package the Codebase

### PR DESCRIPTION
**Pull request by devanshamin**
This PR aims to improve the organization and distribution of the codebase by packaging the BFCL codebase. This PR is part of a series of changes that break down the tasks outlined in #510.

### CODE REVIEW

**Review by CharlieJCJ — APPROVED** [review]
LGTM

File: berkeley-function-call-leaderboard/setup.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    version="0.1.0",
+    description="Berkeley Function Calling Leaderboard (BFCL)",
+    author="Sky Computing Lab",
+    author_email="sky@cs.berkeley.edu",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    python_requires=">=3.10",
+    license="Apache 2.0",
+    packages=find_packages(include=["bfcl*"]),
+    install_requires=[
```
Comment (ShishirPatil) [inline berkeley-function-call-leaderboard/setup.py:18]: Can we actually have this as a part of "requirements.txt" and then read this file to fill in `install_requires`? Or is this way preferred by everyone?

**Review by ShishirPatil — APPROVED** [review]
Hey @devanshamin and @HuanzhiMao , other than the minor comment on requirements, we are good to merge. 
Next steps: 
[ ] Resolve Merge conflicts
[ ] We should update the leader board to reflect #568 #569 and #570 and then tear a release
[ ] Merge this PR

File: berkeley-function-call-leaderboard/setup.py (hunk 1 above)
Comment (devanshamin) [reply berkeley-function-call-leaderboard/setup.py]: Added a `requirements.txt` file

**Review by Fanjia-Yan — APPROVED** [review]
LGTM

**Review by HuanzhiMao — APPROVED** [review]
Good to go

**Comment by ShishirPatil** [comment]
Thank you for the contribution @devanshamin 🚀

**Pull request merged**